### PR TITLE
Added PotionColorCalculationEvent. fires after server calculates poti…

### PR DESCRIPTION
**Pull request by hasunwoo**

### CODE REVIEW

File: src/main/java/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	public boolean isHideParticle() {
+		return hideParticle;
+	}
+
+	public void setHideParticle(boolean hideParticle) {
+		this.hideParticle = hideParticle;
+		isDirty = true;
+	}
+
+	public Collection<PotionEffect> getEffectList() {
```
Comment (liach) [inline src/main/java/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java:67]: Should this be immutable?

File: patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch [hunk 2]
```diff
@@ -39,7 +39,32 @@
...
      public EntityLivingBase func_110144_aD()
-@@ -793,6 +795,8 @@
+@@ -648,9 +650,21 @@
+         }
+         else
+         {
+-            Collection<PotionEffect> collection = this.field_70713_bf.values();
+-            this.field_70180_af.func_187227_b(field_184634_g, Boolean.valueOf(func_184593_a(collection)));
+-            this.field_70180_af.func_187227_b(field_184633_f, Integer.valueOf(PotionUtils.func_185181_a(collection)));
++        	Collection<PotionEffect> collection = this.field_70713_bf.values();
```
Comment (liach) [inline patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch]: tabs should be replaced by 4 spaces.

**Comment by liach** [comment]
You should pr to 1.11.x branch

File: src/main/java/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java (hunk 1 above)
Comment (hasunwoo) [reply src/main/java/net/minecraftforge/event/entity/living/PotionColorCalculationEvent.java:67]: i think it should be. i might have to update my code

**Pull request closed without merging**